syntax = "proto3";

package product;

service ProductService {
  rpc GetProduct (ProductRequest) returns (ProductResponse);
  rpc ListProducts (Empty) returns (ProductListResponse);
}

message ProductRequest {
  string id = 1;
}

message ProductResponse {
  string id = 1;
  string name = 2;
  string description = 3;
  double price = 4;
}

message Product {
  string id = 1;
  string name = 2;
  string description = 3;
  double price = 4;
}

message ProductListResponse {
  repeated Product products = 1;
}

message Empty {}